app/api/study-plan/route.ts

```diff
@@ -59,9 +50,72 @@ export async function GET(req: NextRequest) {
         },
       };
     });
+};
+
+export async function GET(req: NextRequest) {
+  const studyPlanCollaborator = await getCollaborator(req);
+
+  if (!studyPlanCollaborator || studyPlanCollaborator.length === 0) {
+    return NextResponse.json({}, { status: 401 });
+  }
+
+  // TODO: Another todo, we are always taking the studyplanner with index 0, possibly needs be some kind of use preference or last viewed or something
+  const currentCollab = studyPlanCollaborator[0];
+
+  const currentStudyPlan = await getStudyPlanByCollaboratorId(currentCollab.id);
+
+  if (!currentStudyPlan) {
+    return NextResponse.json({}, { status: 401 });
+  }
+
+  /*
+    TODO: maybe add error if list is empty or return something currently if find function errors returns empty list
+    Prev. on error it would automatically return response 500 rn this wouldnt happen 
+    */
+  const semesters = await getSemesterByStudyPlanId(currentStudyPlan.id);
+
+  const mappedSemesters = mapSemster(semesters);
 
-  const res = NextResponse.json({
+  const studyPlan: StudyPlanDTO = {
+    scope: currentStudyPlan.scope,
+    studyPlanCollaborator: currentStudyPlan.studyPlanCollaborator,
     semesters: mappedSemesters,
-  });
-  return res;
+  };
+
+  return NextResponse.json(studyPlan);
+}
+
+export async function PUT(req: NextRequest) {
+  const studyPlanCollaborator = await getCollaborator(req);
+
+  if (
+    !studyPlanCollaborator ||
+    studyPlanCollaborator.length === 0 ||
+    studyPlanCollaborator[0].role != CollaboratorRole.Owner
+  )
+    return NextResponse.json({}, { status: 401 });
+
+  // TODO: Another todo, we are always taking the studyplanner with index 0, possibly needs be some kind of use preference or last viewed or something
+  const currentCollab = studyPlanCollaborator[0];
+  if (currentCollab.role != CollaboratorRole.Owner)
+    return NextResponse.json({}, { status: 401 });
+
+  const body: StudyPlanUpdateScopeDTO = await req.json();
+
+  const updatePlan = await updateStudyPlanScopeByCollabId(
+    currentCollab.id,
+    body,
+  );
+
+  // TODO: Think about better error handling lol
+  if (!updatePlan) return NextResponse.json({}, { status: 500 });
+
+  //TODO: What should the Respone be? return StudyPlan? means the semester needs to be remapped, and if react query, refetches GET /api/study-plan that redundant <- really doesn't make sense to return the studyPlan here I think
```

Comment:
agree, please just return { ok: true } on all PUT and PATCH endpoints and document the reason somewhere.